### PR TITLE
Borovlev Petr

### PR DESCRIPTION
**Pull request by petyb**
All tasks solved

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -4,22 +4,38 @@ import Control.Monad (unless)
...
+                          | planet == "Jupiter" = ageInSeconds / 11.862615 / toYears
+                          | planet == "Saturn" = ageInSeconds / 29.447498 / toYears
+                          | planet == "Uranus" = ageInSeconds / 84.016846 / toYears
+                          | planet == "Neptune" = ageInSeconds / 164.79132 / toYears
+                          | planet == "Pluto" = error "Pluto is not a planet"
+                          | otherwise = error "Give a valid planet name"
 
 isLeapYear :: Int -> Bool
-isLeapYear year = undefined 
+isLeapYear year = (year `mod` 4 == 0 && year `mod` 100 /= 0) || ((year `mod` 400) == 0)
```
Comment (alex28sh) [inline Main.hs:38]: Where is check for negative year?

File: Main.hs [hunk 2]
```diff
@@ -4,22 +4,39 @@ import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+-- short ls = length ls < 3
+short (a : b : c : xs) = False
```
Comment (kajigor) [inline Main.hs:8]: No need to introduce variable names when you are not using them in the body. Use wildcards instead.

File: Main.hs [hunk 3]
```diff
@@ -4,22 +4,39 @@ import Control.Monad (unless)
...
-lovely = undefined 
+lovely a = short a || a!!2 == 14
 
 rightTriangles :: [(Int, Int, Int)]
-rightTriangles = undefined 
+rightTriangles = [ (a, b, c) | c <- [1..], b <- [1..c], a <-[1..b], a^2 + b^2 == c^2]
 
 fizzBuzz :: [String]
-fizzBuzz = undefined 
+fizzBuzz = [s | i <- [1 .. ], let s | i `mod` 3 == 0 && i `mod` 5 == 0 = "FizzBuzz"
```
Comment (kajigor) [inline Main.hs:18]: Don't duplicate code. Introduce a function that checks for divisibility by 3 and 5 respectively, or just check for divisibility by 15.

File: Main.hs [hunk 4]
```diff
@@ -4,22 +4,39 @@ import Control.Monad (unless)
...
-ageOn planet ageInSeconds = undefined 
+ageOn planet ageInSeconds | planet == "Mercury" = ageInSeconds /  0.2408467 / toYears
+                          | planet == "Venus" = ageInSeconds / 0.61519726 / toYears
+                          | planet == "Earth" = ageInSeconds / 1.0 / toYears
+                          | planet == "Mars" = ageInSeconds / 1.8808158 / toYears
+                          | planet == "Jupiter" = ageInSeconds / 11.862615 / toYears
+                          | planet == "Saturn" = ageInSeconds / 29.447498 / toYears
+                          | planet == "Uranus" = ageInSeconds / 84.016846 / toYears
+                          | planet == "Neptune" = ageInSeconds / 164.79132 / toYears
+                          | planet == "Pluto" = error "Pluto is not a planet"
```
Comment (kajigor) [inline Main.hs:34]: You have a lot of code duplication here. It's better to introduce a function for it.

**Comment by kajigor** [comment]
ok

**Pull request closed without merging**